### PR TITLE
feat: create CreatableSelect component

### PR DESCRIPTION
**Pull request by J1za**
## Linear Ticket
https://linear.app/autokitteh/issue/UI-910/creatable-component
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/molecules/select/creatable.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+		{
+			dataTestid,
+			disabled = false,
+			isError = false,
+			label,
+			noOptionsLabel,
+			onChange,
+			onCreateOption,
+			options,
+			placeholder = "Select",
```
Comment (RonenMars) [inline src/components/molecules/select/creatable.tsx:24]: Please remove default placeholder

File: src/constants/featureFlags.constants.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export const featureFlags = {
 	displayDiscordIntegration: import.meta.env.DISPLAY_DISCORD_INTEGRATION,
 	displaySlackSocketIntegration: import.meta.env.DISPLAY_SLACK_SOCKET_INTEGRATION,
+	displayComBoxTrigger: import.meta.env.DISPLAY_COMBOX_TRIGGER,
```
Comment (RonenMars) [inline src/constants/featureFlags.constants.ts:4]:
```suggestion
	displayComboxInTriggersForm: import.meta.env.DISPLAY_COMBOX_IN_TRIGGERS_FORM,
```

File: src/vite-env.d.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@
 // / <reference types="vite-plugin-svgr/client" />
 
 interface ImportMetaEnv {
+	DISPLAY_COMBOX_TRIGGER: any;
```
Comment (RonenMars) [inline src/vite-env.d.ts:6]:
```suggestion
	readonly DISPLAY_COMBOX_IN_TRIGGERS_FORM: boolean;
```

File: vite.config.ts [hunk 4]
```diff
@@ -43,6 +43,7 @@ export default defineConfig({
 		"import.meta.env.VITE_HOST_URL": JSON.stringify(process.env.VITE_HOST_URL),
 		"import.meta.env.DISPLAY_DISCORD_INTEGRATION": process.env.DISPLAY_DISCORD_INTEGRATION,
 		"import.meta.env.DISPLAY_SLACK_SOCKET_INTEGRATION": process.env.DISPLAY_SLACK_SOCKET_INTEGRATION,
+		"import.meta.env.DISPLAY_COMBOX_TRIGGER": process.env.DISPLAY_COMBOX_TRIGGER,
```
Comment (RonenMars) [inline vite.config.ts:46]:
```suggestion
		"import.meta.env. DISPLAY_COMBOX_IN_TRIGGERS_FORM": process.env. DISPLAY_COMBOX_IN_TRIGGERS_FORM,
```

File: src/components/molecules/select/creatable.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
+import React, { forwardRef, useCallback, useEffect, useId, useMemo, useState } from "react";
```
Comment (RonenMars) [inline src/components/molecules/select/creatable.tsx:1]: Why duplicate all of `Select` code?

What do you think about this option?
```
// BaseSelect.tsx
import React, { forwardRef, useCallback, useEffect, useId, useMemo, useState } from "react";
import { useTranslation } from "react-i18next";
import { OptionProps, SingleValue, SingleValueProps, components } from "react-select";
import { SelectOption, SelectProps } from "@interfaces/components";
import { cn } from "@utilities";
import { IconLabel } from "@components/molecules/select";
import { getSelectDarkStyles, getSelectLightStyles } from "@constants";

interface BaseSelectProps extends SelectProps {
  SelectComponent: any; // Type for either ReactSelect or CreatableSelect
}

export const BaseSelect = forwardRef<HTMLDivElement, BaseSelectProps>(
  (
    {
      SelectComponent,
      dataTestid,
      disabled = false,
      isError = false,
      label,
      noOptionsLabel,
      onChange,
      onCreateOption,
      options,
      placeholder = "Select",
      value,
      variant,
      ...rest
    },
    ref
  ) => {
    const [selectedOption, setSelectedOption] = useState<SingleValue<SelectOption>>(null);
    const [isFocused, setIsFocused] = useState(false);
    const { t } = useTranslation("components", { keyPrefix: "select" });
    const { Option, SingleValue } = components;

    useEffect(() => {
      const valueSelected = options.find((option) => option.value === value?.value) || null;
      setSelectedOption(valueSelected);
    }, [value, options]);

    const handleChange = useCallback(
      (selected: SingleValue<SelectOption>) => {
        setSelectedOption(selected);
        onChange?.(selected);
      },
      [onChange]
    );

    const handleFocus = useCallback(() => setIsFocused(true), []);
    const handleBlur = useCallback(() => setIsFocused(false), []);

    const noOptionsMessage = useMemo(() => () => noOptionsLabel || t("noOptionsAvailable"), [noOptionsLabel, t]);
    const selectStyles = useMemo(
      () =>
        variant === "light" ? getSelectLightStyles(isError, disabled) : getSelectDarkStyles(isError, disabled),
      [variant, isError, disabled]
    );

    const labelClass = useMemo(
      () =>
        cn(
          "pointer-events-none absolute -top-1 left-4 text-base opacity-0 transition-all",
          { "-top-2 left-3 px-1 text-xs opacity-100 before:bg-gray-950": isFocused || !!selectedOption },
          {
            "-top-2 left-3 px-1 text-xs opacity-100 before:bg-white":
              (isFocused || !!selectedOption) && variant === "light",
          },
          { "text-gray-900": variant === "light" }
        ),
      [isFocused, selectedOption, variant]
    );

    const borderOverlayLabelClass = useMemo(
      () =>
        cn("absolute left-0 top-1/2 z-0 h-0.5 w-full -translate-y-1/2 bg-black", {
          "bg-white": variant === "light",
        }),
      [variant]
    );

    const id = useId();

    const iconOption = (props: OptionProps<SelectOption>) => {
      const { icon, label } = props.data;
      return (
        <Option {...props}>
          <IconLabel icon={icon} label={label} />
        </Option>
      );
    };

    const iconSingleValue = (props: SingleValueProps<SelectOption>) => {
      const { icon, label } = props.data;
      return (
        <SingleValue {...props}>
          <IconLabel icon={icon} label={label} />
        </SingleValue>
      );
    };

    return (
      <div className="relative" data-testid={dataTestid} ref={ref}>
        <SelectComponent
          {...rest}
          components={{ Option: iconOption, SingleValue: iconSingleValue }}
          id={id}
          isDisabled={disabled}
          isOptionDisabled={(option) => !!option.disabled}
          noOptionsMessage={noOptionsMessage}
          onBlur={handleBlur}
          onChange={handleChange}
          onCreateOption={onCreateOption}
          onFocus={handleFocus}
          options={options}
          placeholder={placeholder}
          styles={selectStyles}
          value={selectedOption}
        />

        <label className={labelClass} htmlFor={id}>
          <span className="relative z-10">{label}</span>
          <span className={borderOverlayLabelClass} />
        </label>
      </div>
    );
  }
);

BaseSelect.displayName = "BaseSelect";

// Select.tsx
import ReactSelect from "react-select";

export const Select = forwardRef<HTMLDivElement, SelectProps>((props, ref) => (
  <BaseSelect {...props} SelectComponent={ReactSelect} ref={ref} />
));

Select.displayName = "Select";

// SelectCreatable.tsx
import CreatableSelect from "react-select/creatable";

export const SelectCreatable = forwardRef<HTMLDivElement, SelectProps>((props, ref) => (
  <BaseSelect {...props} SelectComponent={CreatableSelect} ref={ref} />
));

SelectCreatable.displayName = "SelectCreatable";
```

File: src/components/organisms/triggers/formParts/fileAndFunction.tsx [hunk 6]
```diff
@@ -21,6 +23,25 @@ export const TriggerSpecificFields = ({ filesNameList }: { filesNameList: Select
 	const watchedFunctionName = useWatch({ control, name: "entryFunction" });
 	const watchedEventType = useWatch({ control, name: "eventType" });
 	const watchedFilter = useWatch({ control, name: "filter" });
+	const [selectedOption, setSelectedOption] = useState<SelectOption | null>(null);
```
Comment (RonenMars) [inline src/components/organisms/triggers/formParts/fileAndFunction.tsx:26]: Can we use the form to keep the selected value and not in a state?
For example like connectionType.

File: src/components/molecules/select/creatable.tsx (hunk 5 above)
Comment (J1za) [reply src/components/molecules/select/creatable.tsx:1]: done

File: vite.config.ts (hunk 4 above)
Comment (J1za) [reply vite.config.ts:46]: changed

File: src/vite-env.d.ts (hunk 3 above)
Comment (J1za) [reply src/vite-env.d.ts:6]: changed

File: src/constants/featureFlags.constants.ts (hunk 2 above)
Comment (J1za) [reply src/constants/featureFlags.constants.ts:4]: changed

File: src/components/organisms/triggers/formParts/fileAndFunction.tsx (hunk 6 above)
Comment (J1za) [reply src/components/organisms/triggers/formParts/fileAndFunction.tsx]: changed

**Pull request merged**